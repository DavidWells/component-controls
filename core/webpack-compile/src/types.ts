import * as webpack from 'webpack';
import { RuleTypes } from '@component-controls/webpack-configs';
import { LoadingStore } from '@component-controls/loader';

/**
 * configuration properties for compile and run
 */
export interface CompileProps {
  /**
   * webpack configuration object
   */
  webPack?: webpack.Configuration;
  /**
   * a list of webpack configuration presets from webpack-configs packages
   */
  presets?: RuleTypes;
  /**
   * path to the configuration file e.g : '.storybook'
   */
  configPath?: string;

  /** public output folder for the bundle */
  outputFolder?: string;
}

/**
 * return type from compile and watch functions
 */
export interface CompileResults {
  /**
   * the stories store
   */
  store: LoadingStore;
  /**
   * the webpack stats object
   */
  stats: webpack.Stats;
}

/**
 * adds webpack WatchOptions to the Compiler options
 */
export type WatchProps = {
  watchOptions?: webpack.ICompiler.WatchOptions;
} & CompileProps;
