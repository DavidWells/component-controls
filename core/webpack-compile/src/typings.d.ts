declare module '@component-controls/loader/plugin';
