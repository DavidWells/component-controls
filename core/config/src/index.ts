import * as path from 'path';
import * as fs from 'fs';
import { sync as globSync } from 'glob';
import yargs from 'yargs';
import { Configuration } from '@component-controls/specification';

export const configFileNames = ['main.js', 'main.json', 'main.ts'];

export interface ConfigrationResult {
  config: Configuration;
  configPath: string;
}
/**
 * return the configration object
 * command line accepts -c/ -config parameter for config path
 * the config file is assumed named main.js/main.ts
 * @param baseFolder project folder to start the searh with
 */
export const getConfiguration = (
  baseFolder: string,
  args: string[] = process.argv,
): ConfigrationResult | undefined => {
  const argv = yargs(args)
    .option('config', {
      alias: 'c',
      description: 'configuration folder',
      type: 'string',
    })
    .help()
    .alias('help', 'h').argv;

  const configPath = argv.config
    ? path.resolve(baseFolder, argv.config)
    : baseFolder;
  const configFiles = fs.readdirSync(configPath);
  const configFile = configFiles.find(file =>
    configFileNames.includes(file.toLowerCase()),
  );

  return configFile
    ? {
        config: require(path.resolve(configPath, configFile)),
        configPath,
      }
    : undefined;
};

/**
 * find the story files out of a configuration file
 * using glob for the regex file search
 * @param param0 configuration info return by getConfiguration
 */
export const extractStories = ({
  config,
  configPath,
}: ConfigrationResult): string[] | undefined => {
  const stories =
    config && config.stories
      ? config.stories.reduce((acc: string[], storyRg: string) => {
          const regex = storyRg.replace(/\((.*?)\)/g, '+($1)');
          return [...acc, ...globSync(path.resolve(configPath, regex))];
        }, [])
      : undefined;
  return stories;
};
