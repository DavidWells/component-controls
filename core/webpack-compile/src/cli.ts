import { compile, watch } from './index';
import { cliArgs, CliArgTypes, ArgOptions } from './args';
import { LogLevel } from '@component-controls/logger';

export { cliArgs, defaultCliArgs, ArgOptions } from './args';

export const getArgs = (options?: ArgOptions): CliArgTypes => {
  const argv = cliArgs(options);
  return argv.help().alias('help', 'h').argv;
};
export const run = async (options?: ArgOptions) => {
  const argv = getArgs(options);
  process.env.NODE_ENV = argv.mode;
  const exec = argv.watch ? watch : compile;
  await exec({
    configPath: argv.config,
    presets: argv.presets,
    bundleName: argv.bundle,
    distFolder: argv.dist,
    staticFolder: argv.static,
    logOptions: {
      logLevel: argv.loglevel as LogLevel,
    },
  });
};
