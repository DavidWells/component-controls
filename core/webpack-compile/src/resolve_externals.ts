import { Configuration } from 'webpack';

export interface ResolveExternalsConfig {
  resolve: { alias: { [key: string]: string } };
  externals: Configuration['externals'];
}

export interface ExternalProps {
  lib: string;
  externalName?: string | { [key: string]: string };
}

export class ResolveExternals {
  private _config: ResolveExternalsConfig;
  constructor(config: ResolveExternalsConfig) {
    this._config = config;
  }
  addAlias = (props: ExternalProps): void => {
    const { lib, externalName } = props;
    if (typeof this._config.externals !== 'object') {
      this._config.externals = {};
    }
    (this._config.externals as Record<string, unknown>)[lib] =
      externalName || lib;
  };
}
