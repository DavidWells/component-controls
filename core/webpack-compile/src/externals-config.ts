import { ExternalProps } from './resolve_externals';
export const defaultExternals: ExternalProps[] = [
  {
    lib: 'react',
    externalName: {
      commonjs: 'react',
      commonjs2: 'react',
      amd: 'React',
      root: 'React',
    },
  },
  {
    lib: 'react-dom',
    externalName: {
      commonjs: 'react-dom',
      commonjs2: 'react-dom',
      amd: 'ReactDOM',
      root: 'ReactDOM',
    },
  },
  {
    lib: 'prop-types',
    externalName: {
      commonjs: 'prop-types',
      commonjs2: 'prop-types',
      amd: 'PropTypes',
      root: 'PropTypes',
    },
  },
  { lib: 'regenerator-runtime' },
  {
    lib: 'lodash',
    externalName: {
      commonjs: 'lodash',
      commonjs2: 'lodash',
      amd: 'lodash',
      root: '_',
    },
  },

  { lib: 'theme-ui' },
  { lib: '@theme-ui/presets' },
  { lib: '@theme-ui/core' },
  { lib: '@theme-ui/css' },

  { lib: '@component-controls/core' },
  { lib: '@component-controls/store' },
  { lib: '@component-controls/blocks' },
  { lib: '@component-controls/components' },

  { lib: 'broadcast-channel' },

  { lib: 'react-helmet' },
  { lib: 'react-table' },
  { lib: 'react-colorful' },
  { lib: 'react-google-charts' },
  { lib: 'react-tabs' },
  { lib: 'react-switch' },
  { lib: 'react-popper' },
  { lib: 'react-popper-tooltip' },
  { lib: 'react-animate-height' },
  { lib: 'polished' },
  { lib: 'tinycolor2' },

  { lib: 'prism-react-renderer/themes' },
  { lib: 'prism-react-renderer' },

  { lib: '@primer/octicons-react' },
  { lib: '@mdx-js/react' },

  { lib: 'markdown-to-jsx' },

  { lib: 'axe-core' },
  { lib: 'faker/locale/en_US' },
  { lib: 'fuse.js' },
];
