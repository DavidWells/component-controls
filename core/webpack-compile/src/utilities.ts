import webpack, {
  Configuration,
  Compiler,
  Stats,
  HotModuleReplacementPlugin,
} from 'webpack';
import * as path from 'path';
const chalk = require('chalk');
import LoaderPlugin from '@component-controls/loader/plugin';
import {
  mergeWebpackConfig,
  deepMergeWebpackConfig,
} from '@component-controls/webpack-configs';
import { loadConfiguration } from '@component-controls/config';
import { CompileProps, CompileResults } from './types';
import { ResolveExternals, ResolveExternalsConfig } from './resolve_externals';
import { defaultExternals } from './externals-config';

export type CompileRunProps = CompileProps & {
  /**
   * webpack mode
   */
  mode: Configuration['mode'];
};

const createConfig = (options: CompileRunProps): webpack.Configuration => {
  const { webPack, presets, configPath, mode, outputFolder } = options;
  const distFolder = path.resolve(__dirname, '../dist');
  const plugins = [
    new LoaderPlugin({
      config: configPath,
      escapeOutput: mode === 'development',
    }),
    mode === 'development' && new HotModuleReplacementPlugin({}),
  ].filter(Boolean);
  const webpackConfig = mergeWebpackConfig(
    {
      mode,
      entry: {
        stories: require.resolve(
          '@component-controls/loader/story-store-data.js',
        ),
      },
      output: {
        path: distFolder,
        filename: 'bundle.js',
        libraryTarget: 'umd',
        globalObject: 'this',
      },
      resolve: {
        alias: {},
      },
      externals: {},
      plugins,
      ...(webPack || {}),
    },
    presets,
    { outputFolder, distFolder },
  );

  //add all the aliases to avoid double loading of packages
  const alias = new ResolveExternals(webpackConfig as ResolveExternalsConfig);
  defaultExternals.forEach(a => alias.addAlias(a));

  const runConfig = loadConfiguration(process.cwd(), configPath);

  const userWebpackConfig =
    runConfig?.config &&
    (runConfig.config.webpack || runConfig.config.finalWebpack);

  if (!userWebpackConfig) {
    return webpackConfig;
  }
  return typeof userWebpackConfig === 'function'
    ? userWebpackConfig(webpackConfig, options)
    : deepMergeWebpackConfig(webpackConfig, userWebpackConfig);
};

export const runCompiler = (
  run: (
    compiler: Compiler,
    callback: (err: Error, stats: Stats) => void,
  ) => void,

  props: CompileRunProps,
): Promise<CompileResults> => {
  return new Promise((resolve, reject) => {
    //@ts-ignore
    const compiler = webpack(createConfig(props));
    run(compiler, (err, stats) => {
      if (err) {
        console.error(err);
        return reject(err);
      }
      if (stats.hasErrors() || stats.hasWarnings()) {
        const error = stats.toString({
          errorDetails: true,
          warnings: true,
        });
        console.error(error);
        return reject(error);
      }

      const { store } = require('@component-controls/loader/store');
      console.log(
        chalk.bgRgb(244, 147, 66)(`compiled ${store.stores.length} documents`),
        require.resolve('@component-controls/loader/store'),
      );
      resolve({ store, stats });
    });
  });
};
