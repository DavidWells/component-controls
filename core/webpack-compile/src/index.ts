const chalk = require('chalk');
import { CompileProps, CompileResults, WatchProps } from './types';
import { runCompiler } from './utilities';

export { getBundleName, defBundleName } from './utilities';
export * from './types';
/**
 * compile the stories with webpack
 * returns the stories store object
 */
export const compile = ({
  webPack,
  presets,
  configPath,
  staticFolder,
  distFolder,
  bundleName,
}: CompileProps): Promise<CompileResults> => {
  console.log(
    chalk.bgRgb(244, 147, 66)('@start compilation'),
    'optimized build',
  );
  return runCompiler((compiler, callback) => compiler.run(callback), {
    webPack,
    mode: 'production',
    presets,
    configPath,
    staticFolder,
    distFolder,
    bundleName,
  });
};

/**
 * compile the stories with webpack and launch watching for changes
 * returns the stories store object
 */
export const watch = ({
  webPack,
  presets,
  configPath,
  watchOptions,
  staticFolder,
  distFolder,
  bundleName,
}: WatchProps): Promise<CompileResults> => {
  console.log(
    chalk.bgRgb(244, 147, 66)('@start compilation'),
    'development mode watch',
  );
  return runCompiler(
    (compiler, callback) => compiler.watch({ ...watchOptions }, callback),
    {
      webPack,
      mode: 'development',
      presets,
      configPath,
      staticFolder,
      distFolder,
      bundleName,
    },
  );
};
