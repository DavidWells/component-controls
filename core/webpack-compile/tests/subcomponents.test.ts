import * as path from 'path';
import { compile } from '../src/index';

expect.addSnapshotSerializer({
  test: val => typeof val === 'object' && val instanceof String,
  print: val => {
    return `"${val.toString().trim()}"`;
  },
});

describe('test subcomponents', () => {
  it('compile', async () => {
    const { store } = await compile({
      presets: ['react', 'react-docgen-typescript'],
      configPath: path.resolve(__dirname, 'fixtures', 'subcomponents'),
    });
    expect(store).toMatchSnapshot();
  }, 30000);
});
