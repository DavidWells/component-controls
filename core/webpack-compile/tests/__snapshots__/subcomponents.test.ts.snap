// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test subcomponents compile 1`] = `
Store {
  "_docpaths": undefined,
  "_homepaths": undefined,
  "buildConfig": Object {
    "categories": Array [
      "author",
      "tags",
    ],
    "pages": Object {
      "author": Object {
        "basePath": "authors/",
      },
      "blog": Object {
        "basePath": "blogs/",
      },
      "page": Object {
        "basePath": "pages/",
      },
      "story": Object {
        "basePath": "docs/",
        "storyPaths": true,
        "tabs": Array [
          Object {
            "route": "page",
          },
          Object {
            "route": "test",
          },
        ],
      },
      "tags": Object {
        "basePath": "tags/",
      },
    },
    "stories": Array [
      "./*.stories.(js|jsx|tsx|mdx)",
    ],
  },
  "buildDocPaths": [Function],
  "buildHomePaths": [Function],
  "components": Object {
    "08a97b28475aff9eba8252eeffddeea5": Object {
      "from": "../../../../../examples/stories/src/components/PropTypesButton",
      "importedName": "Button",
      "imports": Object {
        "polished": Array [
          Object {
            "importedName": "lighten",
            "name": "lighten",
          },
        ],
        "prop-types": Array [
          Object {
            "importedName": "default",
            "name": "PropTypes",
          },
        ],
        "react": Array [
          Object {
            "importedName": "default",
            "name": "React",
          },
        ],
      },
      "info": Object {
        "description": "Button with react PropTypes",
        "displayName": "Button",
        "methods": Array [],
        "props": Object {
          "backgroundColor": Object {
            "defaultValue": "'#fefefe'",
            "description": "Background color",
            "type": Object {
              "name": "string",
              "required": false,
            },
          },
          "children": Object {
            "defaultValue": "'default'",
            "description": "button label",
            "type": Object {
              "name": "string",
              "required": false,
            },
          },
          "color": Object {
            "defaultValue": "'black'",
            "description": "Text color, default black",
            "type": Object {
              "name": "string",
              "required": false,
            },
          },
          "disabled": Object {
            "defaultValue": "false",
            "description": "Boolean indicating whether the button should render as disabled",
            "type": Object {
              "name": "boolean",
            },
          },
          "onClick": Object {
            "defaultValue": "() => {}",
            "description": "onClick handler",
            "type": Object {
              "name": "function",
              "required": false,
            },
          },
          "padding": Object {
            "defaultValue": "5",
            "description": "Numeric  field type",
            "type": Object {
              "name": "number",
              "required": false,
            },
          },
          "style": Object {
            "defaultValue": "{}",
            "description": "Custom styles",
            "type": Object {
              "name": "object",
              "value": Array [],
            },
          },
          "type": Object {
            "defaultValue": "'button'",
            "description": "Button type",
            "type": Object {
              "name": "enum",
              "value": Array [
                Object {
                  "name": "string",
                  "value": "'button'",
                },
                Object {
                  "name": "string",
                  "value": "'reset'",
                },
                Object {
                  "name": "string",
                  "value": "'submit'",
                },
              ],
            },
          },
        },
      },
      "loc": Object {
        "end": Object {
          "column": 1,
          "line": 31,
        },
        "start": Object {
          "column": 22,
          "line": 8,
        },
      },
      "name": "Spinner",
      "package": "6b1e448146504b441f181f0e3fc41fc9",
      "request": "/Users/atanasster/component-controls/examples/stories/src/components/PropTypesButton.js",
      "source": "import React from 'react';
import { lighten } from 'polished';
import PropTypes from 'prop-types';

/**
 * Button with react PropTypes
 */
export const Button = ({
  disabled,
  children,
  onClick,
  style,
  backgroundColor,
  color,
  type,
  padding,
}) => (
  <button
    type={type}
    disabled={disabled}
    onClick={onClick}
    style={{
      ...style,
      backgroundColor,
      color: lighten(disabled ? 0.4 : 0, color),
      padding,
    }}
  >
    {children}
  </button>
);

Button.defaultProps = {
  disabled: false,
  children: 'default',
  onClick: () => {},
  style: {},
  backgroundColor: '#fefefe',
  color: 'black',
  type: 'button',
  padding: 5,
};

Button.propTypes = {
  /**
   * Boolean indicating whether the button should render as disabled
   */
  disabled: PropTypes.bool,

  /**
   * button label
   */
  children: PropTypes.string,

  /**
   * onClick handler
   */
  onClick: PropTypes.func,

  /**
   * Custom styles
   */
  style: PropTypes.shape({}),

  /**
   * Background color
   */
  backgroundColor: PropTypes.string,

  /**
   * Text color, default black
   */
  color: PropTypes.string,

  /**
   * Button type
   */
  type: PropTypes.oneOf(['button', 'reset', 'submit']),

  /**
   * Numeric  field type
   */
  padding: PropTypes.number,
};
",
    },
    "b5fffc2a9e710b29374f13f1d7ef028c": Object {
      "from": "../../../../../examples/stories/src/components/Button",
      "importedName": "Button",
      "imports": Object {
        "react": Array [
          Object {
            "importedName": "default",
            "name": "React",
          },
          Object {
            "importedName": "FC",
            "name": "FC",
          },
        ],
      },
      "info": Object {
        "description": "Button with react Typescript properties",
        "displayName": "Button",
        "methods": Array [],
        "props": Object {
          "backgroundColor": Object {
            "defaultValue": "#fefefe",
            "description": "Background color",
            "parentName": "ButtonProps",
            "type": Object {
              "name": "string",
              "raw": "string",
            },
          },
          "children": Object {
            "defaultValue": "default",
            "description": "button label",
            "parentName": "ButtonProps",
            "type": Object {
              "name": "ReactNode",
              "raw": "ReactNode",
            },
          },
          "color": Object {
            "defaultValue": "black",
            "description": "Text color, default black",
            "parentName": "ButtonProps",
            "type": Object {
              "name": "string",
              "raw": "string",
            },
          },
          "disabled": Object {
            "defaultValue": false,
            "description": "Boolean indicating whether the button should render as disabled",
            "parentName": "ButtonProps",
            "type": Object {
              "name": "boolean",
              "raw": "boolean",
            },
          },
          "onClick": Object {
            "defaultValue": "() => {}",
            "description": "onClick handler",
            "parentName": "ButtonProps",
            "type": Object {
              "name": "() => void",
              "raw": "() => void",
            },
          },
          "padding": Object {
            "defaultValue": 5,
            "description": "Numeric  field type",
            "parentName": "ButtonProps",
            "type": Object {
              "name": "number",
              "raw": "number",
            },
          },
          "style": Object {
            "defaultValue": "{}",
            "description": "Custom styles",
            "parentName": "ButtonProps",
            "type": Object {
              "name": "object",
              "raw": "object",
            },
          },
          "type": Object {
            "defaultValue": "button",
            "description": "Button type",
            "parentName": "ButtonProps",
            "type": Object {
              "name": "enum",
              "raw": "\\"button\\" | \\"reset\\" | \\"submit\\"",
              "value": Array [
                Object {
                  "name": "string",
                  "value": "button",
                },
                Object {
                  "name": "string",
                  "value": "reset",
                },
                Object {
                  "name": "string",
                  "value": "submit",
                },
              ],
            },
          },
        },
      },
      "loc": Object {
        "end": Object {
          "column": 1,
          "line": 71,
        },
        "start": Object {
          "column": 39,
          "line": 48,
        },
      },
      "name": "Button",
      "package": "999878322b2acba3a7ca83bc6972a800",
      "request": "/Users/atanasster/component-controls/examples/stories/src/components/Button.tsx",
      "source": "import React, { FC } from 'react';

interface ButtonProps {
  /**
   *  Boolean indicating whether the button should render as disabled
   */
  disabled?: boolean;

  /**
   * button label
   */
  children?: string;

  /**
   * onClick handler
   */
  onClick?: () => void;

  /**
   * Custom styles
   */
  style?: object;

  /**
   * Background color
   */
  backgroundColor?: string;

  /**
   * Text color, default black
   */
  color?: string;

  /**
   * Button type
   */
  type?: 'button' | 'reset' | 'submit';

  /**
   * Numeric  field type
   */
  padding?: number;
}

/**
 * Button with react Typescript properties
 */
export const Button: FC<ButtonProps> = ({
  disabled,
  children,
  onClick,
  style,
  backgroundColor,
  color,
  type,
  padding,
}) => (
  <button
    type={type}
    disabled={disabled}
    onClick={onClick}
    style={{
      ...style,
      backgroundColor,
      color,
      padding,
    }}
  >
    {children}
  </button>
);

Button.defaultProps = {
  disabled: false,
  children: 'default',
  onClick: () => {},
  style: {},
  backgroundColor: '#fefefe',
  color: 'black',
  type: 'button',
  padding: 5,
};
",
    },
  },
  "config": Object {},
  "getDocPage": [Function],
  "getDocPaths": [Function],
  "getDocs": [Function],
  "getHomePage": [Function],
  "getHomePaths": [Function],
  "getIndexPage": [Function],
  "getUniquesByField": [Function],
  "packages": Object {
    "411a6b6f22151d62603115530afde73f": Object {
      "dependencies": Object {
        "@component-controls/loader": "^1.8.0",
        "@component-controls/webpack-configs": "^1.8.0",
        "chalk": "^4.0.0",
        "memory-fs": "^0.5.0",
        "typescript": "^3.8.3",
        "webpack": "^4.43.0",
      },
      "devDependencies": Object {
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@types/jest": "^25.1.2",
        "@types/memory-fs": "^0.3.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
        "webpack-bundle-analyzer": "^3.7.0",
      },
      "fileHash": "411a6b6f22151d62603115530afde73f",
      "name": "@component-controls/webpack-compile",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/webpack-compile/tests/fixtures/subcomponents/esm.stories.tsx",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.8.0",
    },
    "6b1e448146504b441f181f0e3fc41fc9": Object {
      "dependencies": undefined,
      "devDependencies": undefined,
      "fileHash": "6b1e448146504b441f181f0e3fc41fc9",
      "name": "component-controls-stories",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/examples/stories/src/components/PropTypesButton.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.8.0",
    },
    "999878322b2acba3a7ca83bc6972a800": Object {
      "dependencies": undefined,
      "devDependencies": undefined,
      "fileHash": "999878322b2acba3a7ca83bc6972a800",
      "name": "component-controls-stories",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/examples/stories/src/components/Button.tsx",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.8.0",
    },
    "a3c604b46a9a60e6152bda8549a7f0e8": Object {
      "dependencies": Object {
        "@component-controls/loader": "^1.8.0",
        "@component-controls/webpack-configs": "^1.8.0",
        "chalk": "^4.0.0",
        "memory-fs": "^0.5.0",
        "typescript": "^3.8.3",
        "webpack": "^4.43.0",
      },
      "devDependencies": Object {
        "@component-controls/ts-markdown-docs": "^1.5.0",
        "@types/jest": "^25.1.2",
        "@types/memory-fs": "^0.3.2",
        "cross-env": "^5.2.1",
        "eslint": "^6.5.1",
        "jest": "^24.9.0",
        "webpack-bundle-analyzer": "^3.7.0",
      },
      "fileHash": "a3c604b46a9a60e6152bda8549a7f0e8",
      "name": "@component-controls/webpack-compile",
      "peerDependencies": undefined,
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/webpack-compile/tests/fixtures/subcomponents/mdx.stories.mdx",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
      },
      "version": "1.8.0",
    },
  },
  "stores": Array [
    Object {
      "doc": Object {
        "components": Object {},
        "componentsLookup": Object {
          "Button": "b5fffc2a9e710b29374f13f1d7ef028c",
          "Spinner": "08a97b28475aff9eba8252eeffddeea5",
        },
        "date": 2020-07-13T13:59:46.722Z,
        "dateModified": 2020-07-13T13:59:46.724Z,
        "fileName": "/Users/atanasster/component-controls/core/webpack-compile/tests/fixtures/subcomponents/esm.stories.tsx",
        "package": "411a6b6f22151d62603115530afde73f",
        "title": "Library/Components/Button",
      },
      "filePath": "/Users/atanasster/component-controls/core/webpack-compile/tests/fixtures/subcomponents/esm.stories.tsx",
      "stories": Object {
        "overview": Object {
          "arguments": Array [],
          "component": "Button",
          "id": "overview",
          "loc": Object {
            "end": Object {
              "column": 55,
              "line": 9,
            },
            "start": Object {
              "column": 24,
              "line": 9,
            },
          },
          "name": "overview",
          "source": "() => <Button>click me</Button>",
          "subcomponents": Object {
            "Custom tab": "Spinner",
          },
        },
      },
    },
    Object {
      "doc": Object {
        "componentsLookup": Object {
          "Button": "b5fffc2a9e710b29374f13f1d7ef028c",
          "Spinner": "08a97b28475aff9eba8252eeffddeea5",
        },
        "date": 2020-07-13T13:59:53.358Z,
        "dateModified": 2020-07-16T20:43:10.375Z,
        "fileName": "/Users/atanasster/component-controls/core/webpack-compile/tests/fixtures/subcomponents/mdx.stories.mdx",
        "package": "a3c604b46a9a60e6152bda8549a7f0e8",
        "title": "Library/Components/Spinner",
      },
      "filePath": "/Users/atanasster/component-controls/core/webpack-compile/tests/fixtures/subcomponents/mdx.stories.mdx",
      "stories": Object {
        "overview": Object {
          "component": "Button",
          "id": "overview",
          "loc": Object {
            "end": Object {
              "column": 6,
              "line": 31,
            },
            "start": Object {
              "column": 7,
              "line": 29,
            },
          },
          "name": "overview",
          "source": "<Button>click me</Button>",
          "subcomponents": Object {
            "Custom tab": "Spinner",
          },
        },
      },
    },
  ],
}
`;
